### PR TITLE
Introduce API Server to dbs-cli

### PR DESCRIPTION
**Pull request by studychao**
```

    cli: add api server
    
    api server and api client are added to the dbs-cli. Through API server,
    users could send commands when VM is running, so that we could use
    dbs-cli to test operations like device hotplug.
    
    Besides, since api_server and cli_instance share lots of common
    functions so a new trait VmmComm is introduced to decreate the
    repetitive rate of the same code.
    More information could be found in README file and args.md.
    
    Signed-off-by: Chao Wu <chaowu@linux.alibaba.com>
```

```
log: set up log only when command is create
    
    Since now we have 2 commands, we should only set up log only when
    command is create so that when the command is connect the previous log
    won't get flushed.
    
    Signed-off-by: Chao Wu <chaowu@linux.alibaba.com>
```

### CODE REVIEW

File: src/api_client.rs [hunk 1]
```diff
@@ -0,0 +1,38 @@
+// Copyright (C) 2022 Alibaba Cloud. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{Context, Result};
+use serde_json::{json, Value};
+
+use crate::parser::DBSArgs;
+use std::os::unix::net::UnixStream;
+
+// src/bin/server.rs
```
Comment (wllenyj) [inline src/api_client.rs]: remove?

File: src/api_client.rs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+use std::io::Write;
```
Comment (wllenyj) [inline src/api_client.rs]: adjust the order

File: src/main.rs [hunk 3]
```diff
@@ -6,38 +6,51 @@ use std::str::FromStr;
...
+
+            let root = slog::Logger::root(
+                Mutex::new(slog_json::Json::default(file).filter_level(log_level)).map(slog::Fuse),
+                o!("version" => env!("CARGO_PKG_VERSION")),
+            );
+
+            let _guard = set_global_logger(root);
+            run_with_cli(args)?;
+        }
+        Some(Commands::Connect) => {
```
Comment (wllenyj) [inline src/main.rs]: Is the `update` more appropriate?

File: src/parser/mod.rs [hunk 4]
```diff
@@ -46,15 +48,40 @@ pub fn run_with_cli(args: DBSArgs) -> Result<i32> {
...
+        .name("set_cfg".to_owned())
         .spawn(move || {
             cli_instance
-                .run_vmm_server(args)
+                .run_vmm_server(clone_args)
                 .expect("Failed to run server.");
         })
         .unwrap();
 
+    println!("[CHAOTEST] 2");
```
Comment (wllenyj) [inline src/parser/mod.rs]: remove ?

**Review by wllenyj — COMMENTED** [review]
Thanks, 
Please check the ci failure.

File: src/parser/mod.rs (hunk 4 above)
Comment (studychao) [reply src/parser/mod.rs]: done

File: src/main.rs (hunk 3 above)
Comment (studychao) [reply src/main.rs]: thanks, changed.

File: src/api_client.rs (hunk 2 above)
Comment (studychao) [reply src/api_client.rs]: done.

File: src/api_client.rs (hunk 1 above)
Comment (studychao) [reply src/api_client.rs]: done.

**Comment by studychao** [comment]
Hi @wllenyj , I have fixed the comments and also the CI errors. Please take a look.

**Review by wllenyj — APPROVED** [review]

**Pull request merged**